cypress/integration/plugins/security-analytics-dashboards-plugin/1_detectors.spec.js

```diff
@@ -3,171 +3,247 @@
+      cy.get('.reviewFieldMappings').should('be.visible');
+      const properties = interception.response.body.response.properties;
+      const unmapped_field_aliases =
+        interception.response.body.response.unmapped_field_aliases
+          .map((field) => [field])
+          .sort()
+          .slice(0, 10);
+
+      Cypress.log({
+        message: `Validate table data - ${message}`,
+      });
+      if (_.isEmpty(properties)) {
+        validatePendingFieldMappingsPanel(unmapped_field_aliases);
+      } else {
+        let items = getMappingFields(properties, [], '');
+        items = items.map((item) => [item.ruleFieldName, item.logFieldName]);
+        validateAutomaticFieldMappingsPanel(items);
+      }
+    });
+  });
+};
 
-  // Select applicable severity levels
-  cy.get(`[data-test-subj="security-levels-combo-box"]`).click({ force: true });
-  cy.contains('1 (Highest)').click({ force: true });
+const editDetectorDetails = (detectorName, panelTitle) => {
+  cy.sa_urlShouldContain('detector-details').then(() => {
+    cy.sa_getElementByText('.euiTitle', detectorName);
+    cy.sa_getElementByText('.euiPanel .euiTitle', panelTitle);
+    cy.sa_getElementByText('.euiPanel .euiTitle', panelTitle)
+      .parent()
+      .siblings()
+      .within(() => cy.get('button').contains('Edit').click());
+  });
+};
 
-  // Continue to next page
-  cy.contains('Next').click({ force: true });
+const validateAutomaticFieldMappingsPanel = (mappings) =>
+  cy.get('.editFieldMappings').within(() => {
+    cy.get('.euiAccordion__triggerWrapper button').then(($btn) => {
+      cy.get($btn).contains(`Automatically mapped fields (${mappings.length})`);
 
-  // Confirm page is reached
-  cy.contains('Review and create');
+      // first check if the accordion is expanded, if not than expand the accordion
+      if ($btn[0].getAttribute('aria-expanded') === 'false') {
+        cy.get($btn[0])
+          .click()
+          .then(() => {
+            cy.sa_getElementByTestSubject('auto-mapped-fields-table')
+              .find('.euiBasicTable')
+              .sa_validateTable(mappings);
+          });
+      }
+    });
+  });
 
-  // Confirm field mappings registered
-  cy.contains('Field mapping');
+const validatePendingFieldMappingsPanel = (mappings) => {
+  cy.get('.editFieldMappings').within(() => {
+    // Pending field mappings
+    cy.sa_getElementByText('.euiTitle', 'Pending field mappings')
+      .parents('.euiPanel')
+      .within(() => {
+        cy.sa_getElementByTestSubject('pending-mapped-fields-table')
+          .find('.euiBasicTable')
+          .sa_validateTable(mappings);
+      });
+  });
+};
 
-  if (!expectFailure) {
-    for (let field in testMappings.properties) {
-      const mappedTo = testMappings.properties[field].path;
+const fillDetailsForm = (detectorName, dataSource) => {
+  getNameField().type(detectorName);
+  getDataSourceField().sa_selectComboboxItem(dataSource);
+  getDataSourceField().blur();
```

Comment:
cypress complains to me because it's needs to focus the element before blurring

Reply:
my bad, the latest commit didn't get pushed. It should run correctly now